lib/ansible/modules/cloud/ovirt/ovirt_host_storage_facts.py

```diff
@@ -0,0 +1,166 @@
+short_description: Retrieve facts about one or more oVirt/RHV HostStorages (applicable only for block storage)
+author: "Daniel Erez (@derez)"
+version_added: "2.4"
+description:
+    - "Retrieve facts about one or more oVirt/RHV HostStorages (applicable only for block storage)."
+options:
+    host:
+        description:
+            - "Host to get device list from."
+        required: true
+    iscsi:
+        description:
+            - "Dictionary with values for iSCSI storage type:"
+            - "C(address) - Address of the iSCSI storage server."
+            - "C(target) - The target IQN for the storage device."
+            - "C(username) - A CHAP user name for logging into a target."
+            - "C(password) - A CHAP password for logging into a target."
+extends_documentation_fragment: ovirt_facts
+'''
+
+EXAMPLES = '''
+# Examples don't contain auth parameter for simplicity,
+# look at ovirt_auth module to see how to reuse authentication:
+
+# Gather facts about HostStorages with specified target and address:
+- ovirt_host_storage_facts:
+    host: myhost
+    iscsi:
+      target: iqn.2016-08-09.domain-01:nickname
+      address: 10.34.63.204
+- debug:
+    var: ovirt_host_storages
+'''
+
+RETURN = '''
+ovirt_host_storages:
+    description: "List of dictionaries describing the HostStorage. HostStorage attribtues are mapped to dictionary keys,
+                  all HostStorage attributes can be found at following url: http://ovirt.github.io/ovirt-engine-api-model/master/#types/host_storage."
+    returned: On success.
+    type: list
+'''
+
+
+import traceback
+
+try:
+    import ovirtsdk4.types as otypes
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ovirt import (
+    check_sdk,
+    create_connection,
+    get_dict_of_struct,
+    ovirt_facts_full_argument_spec,
+    get_id_by_name,
+)
+
+
+def _login(host_service, iscsi):
+    host_service.iscsi_login(
+        iscsi=otypes.IscsiDetails(
+            username=iscsi.get('username'),
+            password=iscsi.get('password'),
+            address=iscsi.get('address'),
+            target=iscsi.get('target'),
+        ),
+    )
+
+
+def _get_storage_type(params):
+    for sd_type in ['iscsi', 'fcp']:
+        if params.get(sd_type) is not None:
+            return sd_type
+
+
+def main():
+    argument_spec = ovirt_facts_full_argument_spec(
+        host=dict(required=True),
+        iscsi=dict(default=None, type='dict'),
+        fcp=dict(default=None, type='dict'),
+    )
+    module = AnsibleModule(argument_spec)
+    check_sdk(module)
+
+    try:
+        auth = module.params.pop('auth')
+        connection = create_connection(auth)
+
+        # Get Host
+        hosts_service = connection.system_service().hosts_service()
+        host_id = get_id_by_name(hosts_service, module.params['host'])
+        if host_id is None:
+            raise Exception("Host '%s' was not found." % host_id)
```

Comment:
you can remove line 124 and 125, the check is done by `get_id_by_name` function, if host is not found.